likecoin/admin/restful.php

```diff
@@ -45,6 +45,202 @@ function likecoin_rest_refresh_publish_status( $request ) {
+	// phpcs:disable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$boundary = base64_encode( wp_generate_password( 24 ) );
+	// phpcs:enable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$body  = '';
+	$body .= '--' . $boundary . "\r\n";
+	$body .= 'Content-Disposition: form-data; name="index.html"; filename="' . $filename . "\"\r\n";
+	$body .= 'Content-Type: ' . $file_mime_type . "\r\n";
+	$body .= "Content-Transfer-Encoding: binary\r\n";
+	$body .= "\r\n";
+	$body .= $content . "\r\n";
+	$body .= "\r\n";
+	foreach ( $urls as $url ) {
+		$file_info  = new finfo( FILEINFO_MIME_TYPE );
+		$image_path = substr( $url, 22 );
+		$img_body   = file_get_contents( $image_path );
+		$mime_type  = $file_info->buffer( $img_body );
+		$filename   = basename( $url );
+		$body      .= '--' . $boundary . "\r\n";
+		$body      .= 'Content-Disposition: form-data; name="' . $filename . '"; filename="' . $filename . "\"\r\n";
+		$body      .= 'Content-Type: ' . $mime_type . "\r\n";
+		$body      .= "Content-Transfer-Encoding: binary\r\n";
+		$body      .= "\r\n";
+		$body      .= $img_body . "\r\n";
+		$body      .= "\r\n";
+	}
+	$body                   .= '--' . $boundary . '--';
+	$likecoin_api_upload_url = 'https://api.like.co/api/arweave/upload?txHash=' . $tx_hash; // TODO: change based on test/main net.
+	$response                = wp_remote_post(
+		$likecoin_api_upload_url,
+		array(
+			'headers' => array(
+				'Content-Type' => 'multipart/form-data; boundary=' . $boundary,
+			),
+			'body'    => $body,
+		)
+	);
+	if ( is_wp_error( $response ) ) {
+		$err_message = $response->get_error_message();
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			error_log( $response->get_error_message() );
+			$error_response = $response->get_error_message();
+		}
+		return array( 'error' => $response->get_error_message() );
+	}
+	$decoded_response = json_decode( $response['body'], true );
+	if ( ! $decoded_response ) {
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			error_log( $response['body'] );
+		}
+		return array( 'error' => $response['body'] );
+	}
+	$data = json_decode( $response['body'], true );
+	// save ipfsHash to wordpress DB if it does not exist yet. (It could exist if matters info exists)
+	$ipfs_hash = '';
+	$ipfs_info = get_post_meta( $post_id, LC_IPFS_INFO, true );
+	if ( ! is_array( $ipfs_info ) ) {
+		$ipfs_info = array();
+		$ipfs_info['ipfs_hash'] = $data['ipfsHash'];
+		update_post_meta( $post_id, LC_IPFS_INFO, $ipfs_info );
+		$ipfs_hash = $data['ipfsHash'];
+	} else {
+		$ipfs_hash = $ipfs_info['ipfs_hash'];
+	}
+
+	// save arweaveId to wordpress DB.
+	$arweave_info = get_post_meta( $post_id, LC_ARWEAVE_INFO, true );
+	if ( ! is_array( $arweave_info ) ) {
+		$arweave_info = array();
+	}
+	$arweave_info['arweave_id'] = $data['arweaveId'];
+	update_post_meta( $post_id, LC_ARWEAVE_INFO, $arweave_info );
+
+	$data['ipfsHash'] = $ipfs_hash; // ipfs hash returned for ISCN registration.
```

Comment:
I think we also need to register arweaveId for ISCN registration, does current web widget support this?

Reply:
Adding the support here:
https://github.com/likecoin/like-co/pull/1565